subprojects/parseq/src/test/java/com/linkedin/parseq/retry/TestRetryPolicy.java

```diff
@@ -35,6 +37,20 @@ public void testSimpleRetryPolicy()
     assertEquals(task.getError().getMessage(), "current attempt: 2");
   }
 
+  @Test
+  public void testRateLimitedRetryPolicy() throws InterruptedException
+  {
+    RateLimiter rateLimiter = new GuavaRateLimiter(4.0);
+    // wait for the rate limiter to be ready
```

Comment:
why is this needed?

Reply:
Guava rate limiter has a warmup period, and it tries to make the permits "distributed smoothly with the delay between individual permits being adjusted to ensure that the configured rate is maintained"
so that sleep is just for the rate limit to warmup.
The guava implementation has more details https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/util/concurrent/RateLimiter.html#create(double) .